fiftyone/core/utils.py

```diff
@@ -3168,6 +3168,12 @@ def async_executor(
     Raises:
         Exception: if a task raises an exception and ``skip_failures == False``
     """
+    if max_workers is None:
+        max_workers = (
+            fo.config.default_thread_pool_workers
```

Comment:
from the data, it looks like there is no benefit after 8 (most the benefits are seen with 4) so it seems like we should cap it rather than use the default (cpu_count()). For high cpu machines, this could have deleterious affects.